R/dtc_create_iso8601.R

```diff
@@ -0,0 +1,405 @@
+#'   `...`, and so on.
+#' @param .na A character vector of string literals to be regarded as missing
+#'   values during parsing.
+#' @param .cutoff_2000 An integer value. Two-digit years smaller or equal to
+#'   `.cutoff_2000` are parsed as though starting with `20`, otherwise parsed as
+#'   though starting with `19`.
+#' @param .check_format Whether to check the formats passed in `.format`,
+#'   meaning to check against a selection of validated formats in
+#'   [dtc_formats][sdtm.oak::dtc_formats]; or to have a more permissible
+#'   interpretation of the formats.
+#'
+#' @examples
+#' # Converting dates
+#' create_iso8601(c("2020-01-01", "20200102"), .format = "y-m-d")
+#' create_iso8601(c("2020-01-01", "20200102"), .format = "ymd")
+#' create_iso8601(c("2020-01-01", "20200102"), .format = list(c("y-m-d", "ymd")))
+#'
+#' # Two-digit years are supported
+#' create_iso8601(c("20-01-01", "200101"), .format = list(c("y-m-d", "ymd")))
+#'
+#' # `.cutoff_2000` sets the cutoff for two-digit to four-digit year conversion
+#' # Default is at 68.
+#' create_iso8601(c("67-01-01", "68-01-01", "69-01-01"), .format = "y-m-d")
+#'
+#' # Change it to 80.
+#' create_iso8601(c("79-01-01", "80-01-01", "81-01-01"), .format = "y-m-d", .cutoff_2000 = 80)
+#'
+#' # Converting times
+#' create_iso8601("15:10", .format = "HH:MM", .check_format = FALSE)
+#' create_iso8601("2:10", .format = "HH:MM", .check_format = FALSE)
+#' create_iso8601("2:1", .format = "HH:MM", .check_format = FALSE)
+#' create_iso8601("02:01:56", .format = "HH:MM:SS", .check_format = FALSE)
+#' create_iso8601("020156.5", .format = "HHMMSS", .check_format = FALSE)
+#'
+#' # Converting date-times
+#' create_iso8601("12 NOV 202015:15", .format = "dd mmm yyyyHH:MM", .check_format = FALSE)
+#'
+#' # Indicate allowed missing values to make the parsing pass
+#' create_iso8601("U DEC 201914:00", .format = "dd mmm yyyyHH:MM", .check_format = FALSE)
+#' create_iso8601("U DEC 201914:00", .format = "dd mmm yyyyHH:MM", .check_format = FALSE, .na = "U")
+#'
+#' create_iso8601("NOV 2020", .format = "m y", .check_format = FALSE)
+#' create_iso8601(c("MAR 2019", "MaR 2020", "mar 2021"), .format = "m y", .check_format = FALSE)
+#'
+#' create_iso8601("2019-04-041045-", .format = "yyyy-mm-ddHHMM-", .check_format = FALSE)
+#'
+#' create_iso8601("20200507null", .format = "ymd(HH:MM:SS)", .check_format = FALSE)
+#' create_iso8601("20200507null", .format = "ymd((HH:MM:SS)|null)", .check_format = FALSE)
+#'
+#' # Fractional seconds
+#' create_iso8601("2019-120602:20:13.1230001", .format = "y-mdH:M:S", .check_format = FALSE)
+#' @export
+create_iso8601 <- function(..., .format, .na = NULL, .cutoff_2000 = 68L, .check_format = TRUE) {
+
+  dots <- rlang::dots_list(...)
+
+  if (rlang::is_empty(dots)) return(character())
+
+  # Check if all vectors in `dots` are of character type.
+  if (!identical(unique(sapply(dots, typeof)), "character"))
+    rlang::abort("All vectors in `...` must be of type character.")
```

Comment:
Hi @rammprasad: I am not sure which lower-level and wrapper functions you are referring to. If by lower-level function you mean the `create_iso8601()` function, and some other derivation-function as the wrapper, then the try/catch is to be implemented at the wrapper function body. If that is the case, then there is nothing to be done on my end here. Otherwise, if you are suggesting to add the try/catch block around the `rlang::abort()` command, then we are better off not aborting in the first place and decide on a reasonable return value for `create_iso8601()` when this condition happens.

Reply:
This issue about the try/catch block is to be addressed in the appropriate wrapper function, so I am closing this conversation.